pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -183,7 +198,52 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 	}
 	sst.NumStats.Sum = sum
 
-	return &sst, nil
+	if readNumericCount {
+		idx += 8
+		// read NumericCount
+		sst.NumStats.NumericCount = toputils.BytesToUint64LittleEndian(fdata[idx : idx+8])
+	}
+}
+
+func readCurrentSST(sst *structs.SegStats, fdata []byte, idx uint32) error {
+	if sst.IsNumeric {
+		readNumericStats(sst, fdata, idx, true)
+		return nil
+	}
+
+	dType := utils.SS_DTYPE(fdata[idx : idx+1][0])
+	idx += 1
+	if dType != utils.SS_DT_STRING {
+		return fmt.Errorf("readCurrentSST: invalid dtype: %v", dType)
+	}
+
+	// read Min
```

Comment:
nit: remove this comment, this line does not read mean, but there is a below comment that actually reads various things related to min

Reply:
addressed in the latest commit.